packages/contracts/test/utils/combinatorial_utils.ts

```diff
@@ -0,0 +1,119 @@
+import { BigNumber } from '@0xproject/utils';
+import * as combinatorics from 'js-combinatorics';
+
+import { testWithReferenceFuncAsync } from './test_with_reference';
+
+// A set of values corresponding to the uint256 type in Solidity. This set
+// contains some notable edge cases, including some values which will overflow
+// the uint256 type.
+export const uint256Values = [
+    new BigNumber(0),
+    new BigNumber(1),
+    new BigNumber(2),
+    // Non-trivial big number.
+    new BigNumber(2).pow(64),
+    // Max that does not overflow when squared.
+    new BigNumber(2).pow(128).minus(1),
+    // Min that does overflow when squared.
+    new BigNumber(2).pow(128),
+    // Max that does not overflow when doubled.
+    new BigNumber(2).pow(255).minus(1),
+    // Min that does overflow when doubled.
+    new BigNumber(2).pow(255),
+    // Max that does not overflow.
+    new BigNumber(2).pow(256).minus(1),
+    // Min that does overflow.
+    new BigNumber(2).pow(256),
+];
+
+// A set of values corresponding to the bytes32 type in Solidity. This set
+// contains some notable edge cases, including some values which will overflow
+// the bytes32 type.
+export const bytes32Values = [
+    // Min
+    '0x00000000000000000000000000000000',
```

Comment:
These are all only 16 bytes.

Reply:
Nice catch. When I updated the values here it exposed an interesting edge case in `publicUpdateFilledState`. It appears to always return `0` if `orderHash` overflows the `bytes32` type. (Previously, I *though* I was overflowing the `bytes32` type but the input value didn't actually have enough bytes).

Effectively this means that when you call the internal function `updateFilledState` it does not work right if `orderHash` overflows `bytes32`. This might be fine since it's an internal function and we would never expect this to happen. What do you think?